collector/lib/NetworkStatusNotifier.h

```diff
@@ -65,6 +66,7 @@ class NetworkStatusNotifier : protected ProtoAllocator<sensor::NetworkConnection
 
   int64_t afterglow_period_micros_;
   bool enable_afterglow_;
+  const CollectorConfig& config_;
```

Comment:
I am not convinced that a shared-pointer is necessary, because CollectorConfig is stack-allocated in `main()`.

I still introduced shared-pointers, because it will make it easier to change our mind later.

Reply:
> And, the config object was copied before landing in NetworkStatusNotifier. Which would defeat any modification propagation.

Yeah, that's not great, I think with your changes we can delete the copy constructor and assignment operator from `CollectorConfig`, that would be correct if we are considering having a single source of truth for our configuration. We could also remove the move constructor and assignment operators so we can be completely sure no one will be taking ownership of the object away from us.

> I am not convinced that a shared-pointer is necessary, because CollectorConfig is stack-allocated in `main()`.

It's not, but again, single source of truth requires some sort of pointer/singleton/etc..

Reply:
@Molter73, I requested a new review a bit too quickly, sorry. I will remove the move/copy constructors and then let you know when this is ready for a new pass.

Reply:
I gave a bit more thought to this, and I think that using shared_ptr for the config is overkill : if we want to have a singleton at some point, then refcounting is not needed.

I propose to revert my last commit, and simply point to the stack allocated config object.